Description: A number of global variables are declared in headers with
 no explicit linkage.  This results in there being definitions of them
 in multiple object files and causes linking to fail under gcc-10.
 .
 Add `extern` to the header declarations and separate declarations with
 no linkage in files where they are assigned.
Author: Jeremy Sowden <jeremy@azazel.net>
Last-Update: 2020-05-14

--- a/wmclockmon-cal/main.c
+++ b/wmclockmon-cal/main.c
@@ -16,6 +16,8 @@
 #include "main.h"
 
 
+struct tm *timeinfos;
+
 void Usage () {
     printf("Usage: "PACKAGE"-cal [-h] [-v]\n"
            " -h          : short usage help\n"
--- a/wmclockmon-cal/main.h
+++ b/wmclockmon-cal/main.h
@@ -9,7 +9,7 @@
 #include <sys/time.h>
 
 
-struct tm *timeinfos;
+extern struct tm *timeinfos;
 
 void quit_app();
 
--- a/wmclockmon-cal/mainwindow.h
+++ b/wmclockmon-cal/mainwindow.h
@@ -7,20 +7,20 @@
 
 #include <gtk/gtk.h>
 
-GtkWidget *application;
-GtkWidget *alarmlist;
-GtkWidget *wid_backlight;
-GtkWidget *wid_blink;
-GtkWidget *wid_styledir;
-GtkWidget *wid_stylename;
-GtkWidget *wid_color;
-GtkWidget *wid_command;
-GtkWidget *wid_msgcmd;
-GtkWidget *wid_h12;
-GtkWidget *wid_itm;
-GtkWidget *wid_locale;
+extern GtkWidget *application;
+extern GtkWidget *alarmlist;
+extern GtkWidget *wid_backlight;
+extern GtkWidget *wid_blink;
+extern GtkWidget *wid_styledir;
+extern GtkWidget *wid_stylename;
+extern GtkWidget *wid_color;
+extern GtkWidget *wid_command;
+extern GtkWidget *wid_msgcmd;
+extern GtkWidget *wid_h12;
+extern GtkWidget *wid_itm;
+extern GtkWidget *wid_locale;
 
-int selected_row;
+extern int selected_row;
 
 gint list_unsel_cb (GtkCList *clist,
         gint row,
--- a/wmclockmon-config/edit.c
+++ b/wmclockmon-config/edit.c
@@ -11,6 +11,8 @@
 #include "tools.h"
 #include "edit.h"
 
+char *newalarm;
+
 static GtkWidget *wid_status;
 static GtkWidget *wid_time;
 static GtkWidget *wid_date;
--- a/wmclockmon-config/edit.h
+++ b/wmclockmon-config/edit.h
@@ -5,7 +5,7 @@
 #ifndef EDIT_H
 #define EDIT_H
 
-char *newalarm;
+extern char *newalarm;
 
 void edit_dialog(const char *title,
         int on,
--- a/wmclockmon-config/main.c
+++ b/wmclockmon-config/main.c
@@ -13,6 +13,18 @@
 #include "mainwindow.h"
 #include "actions.h"
 
+int    backlight;
+int    switch_authorized;
+int    h12;
+int    use_locale;
+int    showcal;
+int    calalrms;
+char  *style_name;
+char  *style_dir;
+char  *config_file;
+char  *light_color;
+char  *command;
+Alarm *alarms;
 
 void Usage () {
     printf("Usage: "PACKAGE"-config [-h] [-v] [-f filename]\n"
--- a/wmclockmon-config/mainwindow.c
+++ b/wmclockmon-config/mainwindow.c
@@ -10,6 +10,25 @@
 #include "actions.h"
 
 
+GtkWidget *application;
+GtkWidget *alarmlist;
+GtkWidget *wid_backlight;
+GtkWidget *wid_blink;
+GtkWidget *wid_styledir;
+GtkWidget *wid_stylename;
+GtkWidget *wid_color;
+GtkWidget *wid_command;
+GtkWidget *wid_msgcmd;
+GtkWidget *wid_h12;
+GtkWidget *wid_clk;
+GtkWidget *wid_itm;
+GtkWidget *wid_bin;
+GtkWidget *wid_locale;
+GtkWidget *wid_showcal;
+GtkWidget *wid_calalrms;
+
+int selected_row;
+
 static GtkWidget *b_edit;
 static GtkWidget *b_set;
 static GtkWidget *b_remove;
--- a/wmclockmon-config/mainwindow.h
+++ b/wmclockmon-config/mainwindow.h
@@ -7,24 +7,24 @@
 
 #include <gtk/gtk.h>
 
-GtkWidget *application;
-GtkWidget *alarmlist;
-GtkWidget *wid_backlight;
-GtkWidget *wid_blink;
-GtkWidget *wid_styledir;
-GtkWidget *wid_stylename;
-GtkWidget *wid_color;
-GtkWidget *wid_command;
-GtkWidget *wid_msgcmd;
-GtkWidget *wid_h12;
-GtkWidget *wid_clk;
-GtkWidget *wid_itm;
-GtkWidget *wid_bin;
-GtkWidget *wid_locale;
-GtkWidget *wid_showcal;
-GtkWidget *wid_calalrms;
+extern GtkWidget *application;
+extern GtkWidget *alarmlist;
+extern GtkWidget *wid_backlight;
+extern GtkWidget *wid_blink;
+extern GtkWidget *wid_styledir;
+extern GtkWidget *wid_stylename;
+extern GtkWidget *wid_color;
+extern GtkWidget *wid_command;
+extern GtkWidget *wid_msgcmd;
+extern GtkWidget *wid_h12;
+extern GtkWidget *wid_clk;
+extern GtkWidget *wid_itm;
+extern GtkWidget *wid_bin;
+extern GtkWidget *wid_locale;
+extern GtkWidget *wid_showcal;
+extern GtkWidget *wid_calalrms;
 
-int selected_row;
+extern int selected_row;
 
 gint list_unsel_cb (GtkCList *clist,
         gint row,
--- a/wmclockmon-config/tools.c
+++ b/wmclockmon-config/tools.c
@@ -19,6 +19,8 @@
 
 int    free_light_color;
 int    free_command;
+int    time_mode;
+char  *msgcmd;
 
 int fexist(const char *filename) {
     FILE *file;
--- a/wmclockmon-config/variables.h
+++ b/wmclockmon-config/variables.h
@@ -7,22 +7,19 @@
 
 #include "defines.h"
 
-int    backlight;
-int    switch_authorized;
-int    h12;
-int    time_mode;
-int    use_leds;
-int    use_locale;
-int    showcal;
-int    calalrms;
-char  *style_name;
-char  *style_dir;
-char  *config_file;
-char  *light_color;
-char  *command;
-char  *msgcmd;
-Alarm *alarms;
-int    free_light_color;
-int    free_command;
+extern int    backlight;
+extern int    switch_authorized;
+extern int    h12;
+extern int    time_mode;
+extern int    use_locale;
+extern int    showcal;
+extern int    calalrms;
+extern char  *style_name;
+extern char  *style_dir;
+extern char  *config_file;
+extern char  *light_color;
+extern char  *command;
+extern char  *msgcmd;
+extern Alarm *alarms;
 
 #endif
--- a/wmclockmon-cal/mainwindow.c
+++ b/wmclockmon-cal/mainwindow.c
@@ -26,6 +26,8 @@
 #define MONTSTR "XXXX-XX-"
 
 
+GtkWidget *application;
+
 /* Calendar part */
 static GtkWidget *calendar;
 static GtkWidget *closewindow;
